src/components/Header/Header.jsx

```diff
@@ -0,0 +1,22 @@
+import { useState, useEffect } from 'react';
+import NavBar from '@/components/NavigationBar/NavBar';
+
+const Header = () => {
+  const [isvisible, setIsVisible] = useState(true);
+  const [prevScrollPos, setPrevScrollPos] = useState(0);
+
+  useEffect(() => {
```

Comment:
I would go with custom hook for such things..

Reply:
fixed.